src/main/java/org/jabref/logic/bibtex/comparator/BibStringDiff.java

```diff
@@ -88,4 +88,18 @@ public BibtexString getOriginalString() {
     public BibtexString getNewString() {
         return newString;
     }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
```suggestion
    public boolean equals(Object other) {
    
        if (this == other) {
            return true;
        }
        if ((other == null) || (getClass() != other.getClass())) {
            return false;
        }


           BibStringDiff that = (BibStringDiff) other;
        return Objects.equals(newString, other.newString) && Objects.equals(originalString, that.originalString);
```

Do not use toString() beause the BIbTexString class has it's own equals methods which will then be called autoamtically internally.

And tip for the next time: Your IDE has an option to generate equals/hasCode methos

Reply:
That's a good point, but since the BibtexString equals() method is strict and also compares the IDs of two BibtexStrings, it makes things a bit more complicated. 

Is it intended, that two BibtexStrings that only differ in their ID are not considered equal? Because right now these would not be equal:
```
    private final BibtexString bStr1 = new BibtexString("name", "content");
    private final BibtexString bStr2 = new BibtexString("name", "content");
```

Reply:
@koppor you have more knowledge of this BibtexStrings, can I have multiple ones with the same name?

@Davfon you might check the BibtexString dialog for some clues if there are additional restrictions

Reply:
Strings are IMHO key/value pairs: https://docs.jabref.org/fields/strings. 

I think, the Id fiels should not be treated when "equals". However, please, please have a look at the way how a BibEntry is compared and how Ids are treated. There should be a comment on that in the JavaDoc.

Reply:
I think so too (that they are key/value pairs)

A BibEntry is compared by type, fields & commentsBeforeEntry. The Ids are not used for comparing. There was no JavaDoc for the equals method. I would suggest removing the Id comparison in the BibtexString.equals() method. (see last commit)